cfme/services/workloads.py

```diff
@@ -1,23 +1,77 @@
 # -*- coding: utf-8 -*-
 """ A model of Workloads page in CFME
 """
-from functools import partial
-from navmazing import NavigateToSibling, NavigateToAttribute  # NOQA
+from navmazing import NavigateToAttribute
+from widgetastic.widget import View, Text
+from widgetastic_manageiq import Accordion, ManageIQTree, Search
 
-from cfme.exceptions import DestinationNotFound
-from cfme.web_ui import accordion, match_location, toolbar
+from cfme import BaseLoggedInPage
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigator, CFMENavigateStep
 
-vm_instances_tree = partial(accordion.tree, "VMs & Instances", 'All VMs & Instances')
-templates_images_tree = partial(accordion.tree, "Templates & Images", 'All Templates & Images')
 
-# Only this buttons present. may be used in future
-cfg_btn = partial(toolbar.select, 'Configuration')
-pol_btn = partial(toolbar.select, 'Policy')
-lcl_btn = partial(toolbar.select, 'Lifecycle')
+class WorkloadsView(BaseLoggedInPage):
+    search = View.nested(Search)
 
-match_page = partial(match_location, controller='vm_or_template', title='Workloads')
+    def in_workloads(self):
+        return (self.logged_in_as_current_user and
+                self.navigation.currently_selected == ['Services', 'Workloads'])
+
+    @View.nested
+    class vms(Accordion):  # noqa
+        ACCORDION_NAME = "VMs & Instances"
+        tree = ManageIQTree()
+
+        def select_global_filter(self, filter_name):
+            self.tree.click_path("All VMs & Instances", "Global Filters", filter_name)
+
+        def select_my_filter(self, filter_name):
+            self.tree.click_path("All VMs & Instances", "My Filters", filter_name)
+
+        def clear_filter(self):
+            self.parent.search.clear_search()
+            self.tree.click_path("All VMs & Instances")
+
+    @View.nested
+    class templates(Accordion):  # noqa
+        ACCORDION_NAME = "Templates & Images"
+        tree = ManageIQTree()
+
+        def select_global_filter(self, filter_name):
+            self.tree.click_path("All Templates & Images", "Global Filters", filter_name)
+
+        def select_my_filter(self, filter_name):
+            self.tree.click_path("All Templates & Images", "My Filters", filter_name)
+
+        def clear_filter(self):
+            self.parent.search.clear_search()
+            self.tree.click_path("All Templates & Images")
+
+
+class WorkloadsVM(WorkloadsView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            super(WorkloadsVM, self).is_displayed and
```

Comment:
REQUIRED: is_displayed is absent in parent class. I suppose you wanted to call in_workloads.